### ISSUE TITLE
Python error when trying to connect to Talsim-DB from QGIS project containing raster layers

### ISSUE THREAD

**Issue by jamaa** [sydroconsult/QTalsim#3]
Using v1.2.0:

Clicking on "Connect to Talsim DB" produces the following Python error:
```
2024-03-25T08:37:26     WARNING    Traceback (most recent call last):
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim.py", line 2213, in open_sql_connect_dock
              self.sqlConnect = SQLConnectDialog(self.iface.mainWindow(), self)
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim_sqllite_dialog.py", line 28, in __init__
              self.initialize_parameters()
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim_sqllite_dialog.py", line 57, in initialize_parameters
              layers = self.getAllLayers(self.root)
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim.py", line 2032, in getAllLayers
              layers.extend(self.getAllLayers(child))
              File "C:\Users/felixfroehlich/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\QTalsim\qtalsim.py", line 2036, in getAllLayers
              if layer.geometryType() == QgsWkbTypes.PolygonGeometry:
             AttributeError: 'QgsRasterLayer' object has no attribute 'geometryType'
```

I suspect this is because `getAllLayers` is iterating over all layers in the project, and in my case, some of them are raster layers, which behave a bit differently.

**Comment by lorenz3tla** [comment]
Fixed with v1.3.0.

**Comment by lorenz3tla** [comment]
Fixed with v1.3.0.
